File: 2021-09/소수찾기.js
function solution(n) {
    const arr = [...Array(n+1)].map((_, idx) => idx+1);
    let answer = 0;

    for(let i = 2; i <= n; i++) {
        if(arr[i] !== -1) {
            answer++;
            for(let j = i; j <= n; j += i) {
                arr[j] = -1;
            }
        }
    }
    return answer;
}

console.log(solution(10)); // 4

File: programmers/src/main/java/level1/소수찾기.java
package level1;

public class 소수찾기 {

	public int solution(int n) {
		int answer = 0;
		boolean[] prime = new boolean[n + 1];
		for (int i = 2; i <= n; i++) {
			prime[i] = true;
		}

		//제곱근 구하기
		int root = (int) Math.sqrt(n);
		for (int i = 2; i <= root; i++) {
			if (prime[i]) {
				for (int j = i; i * j <= n; j++) {
					prime[i * j] = false;
				}
			}
		}

		for (int i = 2; i <= n; i++) {
			if (prime[i]) {
				answer++;
			}
		}
		return answer;
	}

}
